btf/traversal.go

```diff
@@ -120,15 +132,24 @@ func walkType(typ Type, fn func(*Type)) {
 		fn(&v.Type)
 	case *FuncProto:
 		fn(&v.Return)
-		for i := range v.Params {
-			fn(&v.Params[i].Type)
+		if !reverse {
+			for i := range v.Params {
+				fn(&v.Params[i].Type)
+			}
+		} else {
+			for i := len(v.Params) - 1; i >= 0; i-- {
+				fn(&v.Params[i].Type)
+			}
 		}
 	case *Var:
 		fn(&v.Type)
 	case *Datasec:
-		for i := range v.Vars {
+		for i := len(v.Vars) - 1; i >= 0; i-- {
```

Comment:
Why is this always flipped and not just when reverse=true?

Reply:
Wild guess is this is somehow related to #954, but seems like a curious choice on the surface.

Reply:
My wild guess is that I should've marked this as draft to start with :D